### PR TITLE
ENH: Matplotlib 2.0 compatibility

### LINKED ISSUE #28: Remove matplotlib < 2.0 pin
Can we re-examine the pinning against matplotlib < 2.0? As far as I can tell, the library isn't fundamentally broken with matplotlib. Based on SciTools/Cartopy#773, the only changes needed are to a ticker and to make sure linewidth is set properly. I'm willing to make a patch for conda-forge  for these changes if that helps.

My issue is that in a month I'm teaching a workshop:
1. I will not teach new users matplotlib < 2.0 only to have them turn around and adjust in 6 months
2. CartoPy 0.15.x fixes several bugs I've encountered teaching this very workshop, so using the 0.14.x means I'm walking in with a release with known problems (while the fixes exist in a released version)
3. On conda-forge, dropping back to 0.14.3 also forces me to use numpy 1.11, which means *everything* is using numpy 1.11 (though this hasn't **YET** caused a problem)
4. I'd really like to not have to explain to new users why we need to use two different channels for packages--the whole point of having conda-forge is so we have one community place.

### PR DESCRIPTION
**Pull request by dopplershift**
Pulled in library changes from SciTools/cartopy#733.

Closes #28.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ source:
   sha256: c2221eaf8cb3827b026a398374f3b2e9431aa77fdf95754229232ef5236221ed
   patches:
     - cartopy.win.patch  # [win]
+    # Library changes taken from https://github.com/SciTools/cartopy/pull/773
+    - matplotlib2.patch
 
 build:
   number: 0
```
Comment (ocefpaf) [inline recipe/meta.yaml]: You need to bump the build number.

Comment (dopplershift) [inline recipe/meta.yaml]: Crap. Done.

**Comment by ocefpaf** [comment]
The backlog on Travis-CI is too much and both CircleCI and AppVeyor passed. So I am merging this and monitoring Travis-CI on `master`.

**Comment by dopplershift** [comment]
Great, thanks @ocefpaf !

**Comment by dopplershift** [comment]
Should we kill the pending travis build for this PR?

**Comment by ocefpaf** [comment]
Thanks for the patch. Downgrading `matplotlib` there was quite painful. Now I still need to deal with [this](downgrade) :unamused:

**Comment by ocefpaf** [comment]
> Should we kill the pending travis build for this PR?

I just did that :smile:

**Pull request merged**